mdtraj/formats/hdf5.py

```diff
@@ -314,6 +314,10 @@ def topology(self):
 
         for chain_dict in sorted(topology_dict['chains'], key=operator.itemgetter('index')):
             chain = topology.add_chain()
+            if 'id' in chain_dict.keys():
+                chain.id = chain_dict['id']
+            else:
+                chain.id = chr(chain.index + ord('A'))
```

Comment:
Instead of duplicating this logic here, why don't you re-order things
```python
chain_id = chain_dict['id'] if 'id' in chain_dict else None
chain = topology.add_chain(chain_id)
```